PWGHF/vertexingHF/AliAnalysisTaskHFSimpleVertices.cxx

```diff
@@ -1139,6 +1308,49 @@ Int_t AliAnalysisTaskHFSimpleVertices::DzeroSkimCuts(AliAODRecoDecayHF2Prong* ca
   if(isD0bar) returnValue+=2;
   return returnValue;
  
+}
+//______________________________________________________________________________
+Int_t AliAnalysisTaskHFSimpleVertices::JpsiSkimCuts(AliAODRecoDecayHF2Prong* cand)
+{
+  bool isJpsi = true;
+  Double_t ptCand = cand->Pt();
+  if (ptCand < fJpsiSkimCuts[0]) return 0;
+  Double_t m0=cand->InvMassJPSIee();
+  if(m0 < fJpsiSkimCuts[1] || m0 > fJpsiSkimCuts[2]) isJpsi=false;
+  if (!isJpsi ) return 0;
+  if (cand->CosPointingAngle() < fJpsiSkimCuts[3]) return 0;
+  if (cand->Prodd0d0() > fJpsiSkimCuts[4]) return 0;
+  Int_t returnValue=0;
+  if(isJpsi) returnValue+=1;
+  return returnValue;
+}
+//______________________________________________________________________________
+Int_t AliAnalysisTaskHFSimpleVertices::JpsiSelectionCuts(AliAODRecoDecayHF2Prong* cand,AliESDtrack* trk_p,AliESDtrack* trk_n,AliESDVertex* primvtx,float bzkG)
+{
+  bool isJpsi = true;
+  Double_t dca[2];
+  Double_t dca1[2];
+  Double_t covtrack[2]; Double_t covtrack1[2];
+  Double_t ptCand = cand->Pt();
+  if (ptCand < fMinPtJpsi || ptCand > fMaxPtJpsi) return 0;
+  Int_t jPtBin = GetPtBin(ptCand);
+  if (jPtBin==-1) return 0;
+  
+  if (TMath::Abs(cand->InvMassJPSIee()-fMassJpsi) > fJpsiCuts[jPtBin][0] ) isJpsi=false;
+  if (!isJpsi ) return 0;
```

Comment:
This is an extra if condition that you test and can be avoided by doing:
if (TMath::Abs(cand->InvMassJPSIee()-fMassJpsi) > fJpsiCuts[jPtBin][0] ) return 0;
It seems to me that isJpsi is not needed if you modify in the same way at lines 1343 and 1346.
If you want to preserve isJpsi, you can do:
if (TMath::Abs(cand->InvMassJPSIee()-fMassJpsi) > fJpsiCuts[jPtBin][0] ){
  isJpsi=false;
  return 0;
}

Reply:
Thanks for comments, Yes, It will make the code more clear.